src/bitExpert/PHPStan/Magento/Rules/AbstractModelRetrieveCollectionViaFactoryRule.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the phpstan-magento package.
+ *
+ * (c) bitExpert AG
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace bitExpert\PHPStan\Magento\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleLevelHelper;
+use PHPStan\ShouldNotHappenException;
+use PHPStan\Type\ObjectType;
+
+/**
+ * Since 101.0.0 because collections should be used directly via factory
+ */
+class AbstractModelRetrieveCollectionViaFactoryRule implements Rule
+{
+    /**
+     * @var RuleLevelHelper
+     */
+    private $ruleLevelHelper;
+
+    /**
+     * AbstractModelUseServiceContractRule constructor.
+     *
+     * @param RuleLevelHelper $ruleLevelHelper
+     */
+    public function __construct(RuleLevelHelper $ruleLevelHelper)
+    {
+        $this->ruleLevelHelper = $ruleLevelHelper;
```

Comment:
Looks like this property is unused here. Check out this sniff https://github.com/slevomat/coding-standard#slevomatcodingstandardclassesunusedprivateelements- to detect code like this :)